src/sentry/tempest/tasks.py

```diff
@@ -0,0 +1,178 @@
+    name="sentry.tempest.tasks.fetch_latest_item_id",
+    queue="tempest",
+)
+def fetch_latest_item_id(credentials_id: int) -> None:
+    # FIXME: Try catch this later
+    credentials = TempestCredentials.objects.select_related("project").get(id=credentials_id)
+    project_id = credentials.project.id
+    org_id = credentials.project.organization_id  # this should work?
+    client_id = credentials.client_id
+
+    try:
+        response_text = fetch_latest_id_from_tempest(
+            org_id=org_id,
+            project_id=project_id,
+            client_id=client_id,
+            client_secret=credentials.client_secret,
+        )
+
+        if response_text.isdigit():  # If response is a number the fetch was a success
+            credentials.latest_fetched_item_id = response_text
+            credentials.save(update_fields=["latest_fetched_item_id"])
+            return
+
+        # Things the user ought to know about
+        if response_text.startswith("Invalid credentials"):
+            credentials.message = "Seems like the provided credentials are invalid"
+            credentials.save(update_fields=["message"])
+            return
+        if response_text.startswith("IP address not allow-listed"):
+            credentials.message = "Seems like our IP is not allow-listed"
+            credentials.save(update_fields=["message"])
+            return
+
+        logger.info(
+            "Fetching the latest item id failed.",
+            extra={"org_id": org_id, "project_id": project_id, "client_id": client_id},
+        )
+    except Exception:
+        logger.info(
+            "Fetching the latest item id failed.",
+            extra={"org_id": org_id, "project_id": project_id, "client_id": client_id},
+        )
+
+
+@instrumented_task(
+    name="sentry.tempest.tasks.poll_tempest",
+    queue="tempest",
+)
+def poll_tempest_crashes(credentials_id: int) -> None:
+    # FIXME: Try catch this later
+    credentials = TempestCredentials.objects.select_related("project").get(id=credentials_id)
+    project_id = credentials.project.id
+    org_id = credentials.project.organization_id  # this should work?
+    client_id = credentials.client_id
+
+    try:
+        response_text = fetch_crashes_from_tempest(
+            org_id=org_id,
+            project_id=project_id,
+            client_id=client_id,
+            client_secret=credentials.client_secret,
+            dsn=ProjectKey.objects.get_or_create(
+                use_case=UseCase.TEMPEST, project=credentials.project
+            )[
+                0
+            ].get_dsn(),  # This does generate a dsn not sure if it will work in the grand scheme of things though.
+            offset=int(
+                credentials.latest_fetched_item_id
+            ),  # Need to convert here because it is a char in the DB
+        )
+
+        try:
+            result = json.loads(response_text)
+            credentials.latest_fetched_item_id = result[
+                "latest_id"
```

Comment:
Will this key always be present in the response?

If not, we should check if it exists before accessing it